src/components/user-info/user-info-controller.ts

```diff
@@ -32,6 +34,78 @@ export const userInfoController = async (
     phone_number_verified: config.getPhoneNumberVerified(),
   };
 
+  if (
+    validationResult.claims?.includes("https://vocab.account.gov.uk/v1/address")
+  ) {
+    userInfo["https://vocab.account.gov.uk/v1/address"] =
+      config.getPostalAddressDetails();
+  }
+
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/drivingPermit"
+    )
+  ) {
+    userInfo["https://vocab.account.gov.uk/v1/drivingPermit"] =
+      config.getDrivingPermitDetails();
+  }
+
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/passport"
+    )
+  ) {
+    userInfo["https://vocab.account.gov.uk/v1/passport"] =
+      config.getPassportDetails();
+  }
+
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/socialSecurityRecord"
+    )
+  ) {
+    userInfo["https://vocab.account.gov.uk/v1/socialSecurityRecord"] =
+      config.getSocialSecurityRecordDetails();
+  }
+
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/returnCode"
+    )
+  ) {
+    userInfo["https://vocab.account.gov.uk/v1/returnCode"] =
+      config.getReturnCodes();
+  }
+
+  const signingKey = await importPKCS8(EC_PRIVATE_TOKEN_SIGNING_KEY, "EC");
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/coreIdentityJWT"
+    )
+  ) {
+    const coreIdentity = config.getCoreIdentityDetails();
+    userInfo["https://vocab.account.gov.uk/v1/coreIdentityJWT"] =
+      await new SignJWT(coreIdentity as any)
+        .setProtectedHeader({
+          alg: "ES256",
+        })
+        .sign(signingKey);
+  }
+
+  if (
+    validationResult.claims?.includes(
+      "https://vocab.account.gov.uk/v1/inheritedIdentityJWT"
+    )
+  ) {
+    const inheritedIdentity = config.getInheritedIdentityDetails();
+    userInfo["https://vocab.account.gov.uk/v1/inheritedIdentityJWT"] =
+      await new SignJWT(inheritedIdentity as any)
+        .setProtectedHeader({
+          alg: "ES256",
+        })
+        .sign(signingKey);
+  }
```

Comment:
This is not something that is returned in the user info endpoint. This is a slight abuse of the spec for HMRC to pass us additional information about a user

Reply:
Removed